math/src/elliptic_curve/short_weierstrass/curves/bls12_381/field_extension.rs

```diff
@@ -139,7 +158,19 @@ impl ByteConversion for FieldElement<Degree2ExtensionField> {
     }
 }
 
-///////////////
+/////////////
+
+#[derive(Debug, Clone)]
+pub struct MyQuadraticNonResidue;
+
+impl HasQuadraticNonResidue for MyQuadraticNonResidue {
+    type BaseField = BLS12381PrimeField;
+
+    fn residue() -> FieldElement<Self::BaseField> {
+        -FieldElement::one()
+    }
+}
```

Comment:
Code for testing purposes should go in the `tests` module
```suggestion
```

Reply:
Oh I see now that you were using it for the conjugate function. Not only for testing